frontend/src/components/table/Table.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ReactElement } from "react";
+import Card from "../card/Card";
+import classes from "./Table.module.scss";
+import {
+  flexRender,
+  getCoreRowModel,
+  Header,
+  HeaderGroup,
+  Row,
+  TableOptions,
+  useReactTable,
+} from "@tanstack/react-table";
+import { DecoratedPollingEntity } from "../../hooks/use-timeout-polling";
+import { CommonUtils } from "../../utils/common-utils";
+
+export type CustomTableType = {
+  renderCustomHeader?: (header: HeaderGroup<TableData<any>>) => ReactElement;
+  renderCustomRow?: (row: Row<TableData<any>>) => ReactElement;
+};
+
+export type TableProps<TData> = Pick<
+  TableOptions<TableData<TData>>,
+  "data" | "columns"
+> &
+  CustomTableType;
+
+export type TableData<TData> = DecoratedPollingEntity<TData> | TData;
+
+function Table<TData>({
+  columns,
+  data,
+  renderCustomRow,
+  renderCustomHeader,
+}: TableProps<TData>): ReactElement {
+  const table = useReactTable<TableData<TData>>({
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  return (
+    <div>
+      {table.getHeaderGroups().map((headerGroup) =>
+        renderCustomHeader ? (
+          renderCustomHeader(headerGroup)
+        ) : (
+          <Card
+            className={`${classes.tableRow} ${classes.headerRow}`}
+            key={headerGroup.id}
+            variant="header-row"
+          >
+            {headerGroup.headers.map((header) => (
+              <div key={header.id}>
+                {flexRender(
```

Comment:
Not sure what does this `flexRender` do. I also couldn't find much related documentation, because react-table@v8 docs are still kinda in progress i guess.

Could you describe in a comment why is this used or add a documentation source?